marconi-chain-index/src/Marconi/ChainIndex/Experimental/Indexers/MintTokenEvent.hs

```diff
@@ -520,43 +528,44 @@ instance
-                  [ ":slotNo" := C.chainPointToSlotNo cp
-                  , ":policyId" := policyId
-                  , ":assetName" := assetName
-                  ]
-                QueryByAssetId policyId Nothing _ ->
-                  [ ":slotNo" := C.chainPointToSlotNo cp
-                  , ":policyId" := policyId
-                  ]
-          )
-          ( \(QueryByAssetId _ assetNameM eventType) ->
-              let policyIdWhereClause = Just "policyId = :policyId"
-                  assetNameWhereClause = fmap (const "assetName = :assetName") assetNameM
-                  eventTypeWhereClause =
-                    case eventType of
-                      Nothing -> Nothing
-                      Just MintEventType -> Just "quantity > 0"
-                      Just BurnEventType -> Just "quantity < 0"
-                  whereClause =
-                    fmap (\e -> Text.intercalate " AND " $ NonEmpty.toList e) $
-                      NonEmpty.nonEmpty $
-                        catMaybes [policyIdWhereClause, assetNameWhereClause, eventTypeWhereClause]
-               in mintTokenEventQueryByAssetId whereClause
-          )
-          (const toTimedMintEvents)
+    Core.querySyncedOnlySQLiteIndexerWith
+      ( \cp ->
+          \case
+            QueryByAssetId policyId (Just assetName) _ ->
+              [ ":slotNo" := C.chainPointToSlotNo cp
+              , ":policyId" := policyId
+              , ":assetName" := assetName
+              ]
+            QueryByAssetId policyId Nothing _ ->
+              [ ":slotNo" := C.chainPointToSlotNo cp
+              , ":policyId" := policyId
+              ]
+      )
+      ( \(QueryByAssetId _ assetNameM eventType) ->
+          let policyIdWhereClause = Just "policyId = :policyId"
+              assetNameWhereClause = fmap (const "assetName = :assetName") assetNameM
+              eventTypeWhereClause =
+                case eventType of
+                  Nothing -> Nothing
+                  Just MintEventType -> Just "quantity > 0"
+                  Just BurnEventType -> Just "quantity < 0"
+              whereClause =
+                fmap (\e -> Text.intercalate " AND " $ NonEmpty.toList e) $
+                  NonEmpty.nonEmpty $
+                    catMaybes [policyIdWhereClause, assetNameWhereClause, eventTypeWhereClause]
+           in mkMintTokenEventQueryBy whereClause
+      )
+      (const toTimedMintEvents)
+
+mkMintTokenEventQueryBy :: Maybe Text -> SQL.Query
+mkMintTokenEventQueryBy whereClauseM =
+  SQL.Query $
+    Text.intercalate " " $
+      catMaybes
+        [ Just
+            "SELECT slotNo, blockHeaderHash, blockNo, txId, txIndexInBlock, policyId, assetName, quantity, redeemerHash, redeemerData"
+        , Just "FROM minting_policy_events"
+        , Just "WHERE slotNo <= :slotNo"
+        , fmap (\e -> Text.append " AND " e) whereClauseM
+        , Just "ORDER BY slotNo ASC, blockNo ASC, txIndexInBlock ASC"
+        ]
```

Comment:
```
[sql|
  SELECT
    slotNo,
    blockHeaderHash,
    blockNo,
    txId,
    txIndexInBlock,
    policyId,
    assetName,
    quantity,
    redeemerHash,
    redeemerData
  FROM minting_policy_events
  WHERE slotNo <= :slotNo
]
<> maybe "" (" AND " <> e) whereClauseM
<> [sql|ORDER BY slotNo ASC, blockNo ASC, txIndexInBlock ASC]
```

Reply:
My issue with this approach is the addition of spaces which are hard to spot if an error occurs.
For example, the ORDER BY statement needs a space:

```
[sql| ORDER BY slotNo ASC, blockNo ASC, txIndexInBlock ASC]
```